src/test/kotlin/nl/info/zac/authentication/RequestAuthorizationFilterTest.kt

```diff
@@ -0,0 +1,480 @@
+/*
+ * SPDX-FileCopyrightText: 2025 INFO.nl
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package nl.info.zac.authentication
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.mockk.checkUnnecessaryStub
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.runs
+import io.mockk.verify
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import jakarta.servlet.http.HttpSession
+import nl.info.zac.identity.model.ZACRole
+
+class RequestAuthorizationFilterTest : BehaviorSpec({
+
+    val httpServletRequest = mockk<HttpServletRequest>()
+    val httpServletResponse = mockk<HttpServletResponse>()
+    val filterChain = mockk<FilterChain>()
+    val httpSession = mockk<HttpSession>(relaxed = true)
+
+    beforeEach {
+        checkUnnecessaryStub()
+    }
+
+    fun setSessionUser(user: LoggedInUser?) {
+        if (user == null) {
+            every { httpServletRequest.getSession(false) } returns null
+        } else {
+            every { httpServletRequest.getSession(false) } returns httpSession
+            every { httpSession.getAttribute("logged-in-user") } returns user
+        }
+    }
+
+    Given("POST /rest/notificaties is public") {
```

Comment:
Minor thing but this is not a given really. What is given is a request on "/rest/notificaties" without authentication. The fact that an unauthenticated POST on this URI is allowed is the functionality you are testing; it is not a given in the sense of BDD. I feel we are moving away from how Given-When-Then tests are intended to be written here.

Similar for the other tests. The tests look fine and are very useful, I just think we can be a bit more careful in the descriptions, also to align them with our other tests.

```suggestion
    Given("An unauthenticated POST request on '/rest/notificaties'") {
```